test/test_nn.py

```diff
@@ -1202,16 +1232,17 @@ def test_gumbel_softmax_st(self):
         self._test_gumbel_softmax_st(False)
 
     @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    # torch.nn.functional doesn't provide half() method
```

Comment:
I'm not sure if I understand the comment. It doesn't and it won't ever provide such a method. It should work with all types just fine